viewing_party/party.py

```diff
@@ -1,23 +1,148 @@
 # ------------- WAVE 1 --------------------
 
+from symbol import subscriptlist
+
+
 def create_movie(title, genre, rating):
-    pass
+  new_move = {"title": title,
+              "genre" : genre,
+              "rating": rating}
+  for key, value in new_move.items():
+    if value == None:
+      return None
+  else:
+      return new_move
+
+def add_to_watched(user_data, movie):
+  user_data = {"watched":[movie]}
+  return user_data
+
+def add_to_watchlist(user_data,movie):
+  user_data = {"watchlist":[movie]}
+  return user_data
+
+def watch_movie(user_data, title):
+  for movie in user_data["watchlist"]:
+    if movie['title'] == title:
+      user_data["watched"].append(movie)
+      user_data["watchlist"].remove(movie)
+
+  return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+  movies_watched = user_data["watched"]
+  length = len(movies_watched)
+    
+  total = 0
 
+  for ratings in movies_watched:
+    total += float(ratings["rating"])
+    average = total / length
+  if length == 0:
+    return 0
+  
+  return average
 
+def get_most_watched_genre(user_data):
+  movies = user_data["watched"]
+  genres = {}
+
+  for movie_watched in movies:

```

Comment:
We can directly iterate over user_data["watched"] here if you'd like since the expression isn't too lengthy/complicated:
```python
 for movie_watched in user_data["watched"]:
```